bankroll/converter/modelconverter.py

```diff
@@ -0,0 +1,74 @@
+from typing import Any, Callable, cast, List, Optional, Sequence, Tuple, Union
+
+try:
+    import pandas as pd
+except ImportError:
+    pass
+
+from bankroll.model import Position, Trade
+
+ConvertibleModel = Union[Position, Trade]
+ColumnFunctions = List[Tuple[str, Callable[[Any], Any]]]
+
+
+def dataframeForModelObjects(
+    items: Sequence[ConvertibleModel]
+) -> Optional[pd.DataFrame]:
+    df = None
+
+    if module_exists("pandas"):
+        rows: List[List[Any]] = []
+        firstItem: Optional[ConvertibleModel] = None
+        for i in items:
+            if firstItem == None:
+                firstItem = i
+            else:
+                assert type(firstItem) == type(i)
+
+            rows.append([t[1](i) for t in dataframeColumnFunctions(i)])
```

Comment:
With the change to using a `Dict` and a `TypeVar` as mentioned above, this whole loop could be collapsed into:

```suggestion
            rows = [[fn(i) for fn in dataframeColumnFunctions.values()] for i in items]
```